core/src/main/java/org/apache/calcite/util/format/FormatElementEnum.java

```diff
@@ -217,6 +291,34 @@ public enum FormatElementEnum implements FormatElement {
       sb.append(String.format(Locale.ROOT, "%d", (calendar.get(Calendar.MONTH) / 3) + 1));
     }
   },
+  AMPM("", "The time as Meridian Indicator in uppercase") {
+    @Override public void format(StringBuilder sb, Date date) {
+      final Calendar calendar = Work.get().calendar;
+      calendar.setTime(date);
+      sb.append(calendar.get(Calendar.AM_PM) == Calendar.AM ? "AM" : "PM");
```

Comment:
can you just call `.toString()` on calendar.get(am_pm)` or no

Reply:
Checked, `Calendar` in an INT enum, so the returned value needs to be compared and output as a string explicitly, as the enum only contains its int value.

Reply:
Apologies, not an enum but the class stores these calendar elements as member properties of type int, so the conversion to string is still necessary.